fixes/consolidate-multiple-richtext-history.sql

```diff
@@ -0,0 +1,132 @@
+	join dbo.Audit nextA on nextA.ID = Assets.NextAuditID
+	WHERE
+	ISNULL(currentA.ChangedByID,-1) = ISNULL(nextA.ChangedByID,-1)  -- Consecutive changes from the same user
+	AND DATEDIFF(mi,currentA.ChangeDateUTC,nextA.ChangeDateUTC) <= @timeThreshold --Time threshold / period / lapse.
+	AND	ISNULL(currentAsset.@field,-1) != ISNULL(nextAsset.@field,-1) --@FieldName has changed
+	if (@saveChanges != ''1'') select NULL as ''#suspect'', * from #suspect
+
+	-- consecutive redundant entries
+	insert #bad(ID, CurrentAuditID, NextAuditID)
+	select _.ID, CurrentAuditID, NextAuditID
+	from #suspect _
+	join [@table] A on A.ID=_.ID and A.AuditBegin=_.CurrentAuditID
+	join [@table] B on B.ID=_.ID and B.AuditEnd=_.CurrentAuditID AND ISNULL(B.@field,-1) != ISNULL(A.@field,-1) ' + @colsAB + '
+	if (@saveChanges != ''1'') select NULL as ''#bad'', * from #bad
+
+	-- Rows to purge
+	select NULL as ''will purge'', cAsset.ID, cAsset.AssetType, cAsset.AuditBegin, a.ChangedByID, a.ChangeDateUTC
+	from #bad b
+	join [@table] cAsset ON cAsset.ID=b.ID and cAsset.AuditBegin=b.CurrentAuditID
+	join dbo.Audit a on a.ID = b.CurrentAuditID
+
+	-- Purge
+	delete [@table]
+	from #bad
+	where [@table].ID=#bad.ID and [@table].AuditBegin=#bad.CurrentAuditID
+
+	-- Check for errors
+	declare @error int, @rowcount int
+	select @rowcount=@@ROWCOUNT, @error=@@ERROR
+	if @error<>0 goto ERR
+	raiserror(''%d %s.%s historical records purged'', 0, 1, @rowcount, @tableName, @fldName)
+	if @rowcount=0 goto FINISHED
+
+	--re-stitch Asset history
+	;with H as (
+		select ID, AuditBegin, AuditEnd, R=ROW_NUMBER() over(partition by ID order by AuditBegin)
+		from [@table]
+	)
+	update [@table] set AuditEnd=B.AuditBegin
+	from H A
+	left join H B on A.ID=B.ID and A.R+1=B.R
+	where [@table].ID=A.ID and [@table].AuditBegin=A.AuditBegin
+	and isnull(A.AuditEnd,-1)<>isnull(B.AuditBegin,-1)
+
+	-- sync up Table_Now with history tips from Table
+	alter table dbo.@tblNow disable trigger all
+	update dbo.@tblNow set AuditBegin=[@table].AuditBegin
+	from dbo.[@table]
+	where [@table].ID=@tblNow.ID and [@table].AuditEnd is null and [@table].AuditBegin<>@tblNow.AuditBegin
+	alter table dbo.@tblNow enable trigger all
+
+	-- Log number of _Now records synced
+	select @rowcount=@@ROWCOUNT, @error=@@ERROR
+	if @error<>0 goto ERR
+	raiserror(''%d records syncd'', 0, 1, @rowcount) with nowait
+
+	if @saveChanges = ''1'' begin
+		DBCC DBREINDEX([@table])
+		exec dbo.AssetAudit_Rebuild
+		DBCC DBREINDEX(AssetAudit)
+	end
+
+	FINISHED:
+	if (@saveChanges = ''1'') goto OK
+	raiserror(''Rolling back changes.  To commit changes, set saveChanges="1"'',16,1)
+	ERR: rollback tran TX
+	OK: commit
+
+	drop table #bad
+	drop table #suspect
+	drop table #assets
+'
+declare @sql varchar(max) = @template2
+select @sql = replace(@sql, token, value) from (values
+	('[@table]', quotename(@table)),
+	('@tableName', quotename(@tableName, '''')),
+	('@tblNow', quotename(@tableName + '_Now')),
```

Comment:
```suggestion
	('@tableName', quotename(@table, '''')),
	('@tblNow', quotename(@table + '_Now')),
```

Should @tableName ever be different than @table?